server/repository/src/db_diesel/sync_file_reference_row.rs

```diff
@@ -108,10 +161,29 @@ impl<'a> SyncFileReferenceRowRepository<'a> {
         diesel::update(sync_file_reference.filter(id.eq(sync_file_reference_id)))
             .set(deleted_datetime.eq(Some(chrono::Utc::now().naive_utc())))
             .execute(&self.connection.connection)?;
-        self.insert_changelog(
-            sync_file_reference_id.to_owned(),
-            ChangelogAction::Delete,
-        )?;
+        self.insert_changelog(sync_file_reference_id.to_owned(), ChangelogAction::Delete)?;
+        Ok(())
+    }
+
+    pub fn find_all_to_upload(&self) -> Result<Vec<SyncFileReferenceRow>, RepositoryError> {
+        let result = sync_file_reference
+            .filter(deleted_datetime.is_null())
+            .filter(direction.eq(SyncFileDirection::Upload))
+            .filter(
+                status.eq(SyncFileStatus::New).or(status
```

Comment:
This version lets us do the retry at logic, easily, I find this easier to reason about than `Not Done || Not Permentant`.

I wasn't sure if we wanted to process `InProgress`, However I've convinced myself that if the server is restarted it could be stuck as Inprogress so have updated.